File: 037_bench_with_c/c/main.c
#include <stdio.h>
#include <math.h>


struct benchTest {
  long double value;
};

long double bench(long double f) {
	struct benchTest s;
	s.value = sinl(f);
	return s.value;
}

int main() {
    long double tt=1.0;
    for(int i=0;i<10000000;i++) {
        tt=bench(tt);
    }
    printf("%0.19Lf\n", tt);
}

File: 037_bench_with_c/main.go
package main

import (
	"fmt"
	"math"
)

type benchTest struct {
	value float64
}

func bench(f float64) float64 {
	var s benchTest
	s.value = math.Sin(f)
	return s.value
}

func main() {
	tt := math.Sin(1.0)
	for i := 0; i < 10000000; i++ {
		tt = bench(tt)
	}
	fmt.Println(tt)
}
